Index: private/stf-apps/atwd_pmt_led.c
===================================================================
RCS file: /home/icecube/cvsroot/stf/private/stf-apps/atwd_pmt_led.c,v
retrieving revision 1.4
retrieving revision 1.5
diff -u -r1.4 -r1.5
--- private/stf-apps/atwd_pmt_led.c	20 Jul 2004 20:32:45 -0000	1.4
+++ private/stf-apps/atwd_pmt_led.c	22 Dec 2004 01:29:52 -0000	1.5
@@ -149,8 +149,8 @@
 	}
 	
 	for (l=2; l<cnt; l++) {
-	  if ((light_pulse_waveform[l]-light_pulse_waveform[l-2])>10 && 
-	      (light_pulse_waveform[l-1]-light_pulse_waveform[l-2])>10)
+	  if ((light_pulse_waveform[l]-light_pulse_waveform[l-2])<-10 && 
+	      (light_pulse_waveform[l-1]-light_pulse_waveform[l-2])<-10)
 	    {
 	      count++;
 	      break;
@@ -270,5 +270,9 @@
    free(buffer2);
    return       
      *real_hv_output > 0.95*pmt_hv_high_volt &&
-     *real_hv_output < 1.05*pmt_hv_high_volt;
+     *real_hv_output < 1.05*pmt_hv_high_volt &&
+     ((LED_dac==0 && *light_pulse_count>=(loop_count*0.95)) || (LED_dac==1023 && *light_pulse_count<=(loop_count*0.05))) &&
+     ((LED_dac==0 && *atwd_waveform_amplitude>880) || (LED_dac==1023 && *atwd_waveform_amplitude<2)) &&
+     ((LED_dac==0 && *LED_waveform_amplitude>60) || (LED_dac==1023 && *LED_waveform_amplitude<2)) &&
+     (LED_dac==0 && *LED_waveform_position>=38 && *LED_waveform_position<=44);
 }
