--- public/hal/DOM_MB_fpga.h	24 Nov 2004 19:47:10 -0000	1.40.2.2
+++ public/hal/DOM_MB_fpga.h	16 Jan 2005 23:53:54 -0000	1.44
@@ -140,6 +140,18 @@
 void
 hal_FPGA_TEST_trigger_disc(int trigger_mask);
 
+/**
+ * discriminator launch of the atwd/fadc...
+ * when LC is enabled
+ * \param trigger_mask bitmask of devices to trigger
+ *
+ * \see HAL_FPGA_TEST_TRIGGER_ATWD0
+ * \see HAL_FPGA_TEST_TRIGGER_ATWD1
+ * \see HAL_FPGA_TEST_TRIGGER_FADC
+ */
+void
+hal_FPGA_TEST_trigger_disc_lc(int trigger_mask);
+
