diff -rNu stfapp/src/icecube/daq/stf/stfapp/RunSummary.java ../icecube/stfapp/src/icecube/daq/stf/stfapp/RunSummary.java
--- stfapp/src/icecube/daq/stf/stfapp/RunSummary.java	Thu Jan  1 01:00:00 1970
+++ stfapp/src/icecube/daq/stf/stfapp/RunSummary.java	Tue Feb  8 00:35:16 2005
@@ -0,0 +1,46 @@
+/*************************************************  120 columns wide   ************************************************
+
+ Class:  	RunSummary
+
+ @author 	Jim Braun
+ @author     jbraun@amanda.wisc.edu
+
+ ICECUBE Project
+ University of Wisconsin - Madison
+
+ **********************************************************************************************************************/
+
+package icecube.daq.stf.stfapp;
+
+import java.io.OutputStream;
+import java.io.PrintWriter;
+import java.util.GregorianCalendar;
+import java.util.TimeZone;
+import java.text.DateFormat;
+
+public class RunSummary {
+
+    private PrintWriter out;
+    private static DateFormat format = DateFormat.getDateTimeInstance(DateFormat.SHORT, DateFormat.LONG);
+
+    public RunSummary(OutputStream outstr) {
+        out = new PrintWriter(outstr, true);
+    }
+
+    public void appendResult(String domId, String name, boolean pass) {
+        StringBuffer str1 = new StringBuffer(100);
+        for (int i = 0; i < 80; i++) str1.insert(i, " ");
+        String dstr = format.format((new GregorianCalendar(TimeZone.getTimeZone("GMT"))).getTime());
+        str1.replace(0, dstr.length(), dstr);
+        str1.replace(25, 25 + domId.length(), domId);
+        str1.replace(40, 40 + name.length(), name);
+        str1.replace(99, 100, pass ? "P" : "F");
+        out.println(str1);
+        out.flush();
+    }
+
+    public void close() {
+        out.flush();
+        out.close();
+    }
+}
