echo-mode is now echo-mode again, not domcal!

Index: iceboot/resources/startup.fs
===================================================================
RCS file: /home/icecube/cvsroot/iceboot/resources/startup.fs,v
retrieving revision 1.23.2.1
diff -u -r1.23.2.1 startup.fs
--- iceboot/resources/startup.fs	10 Feb 2005 14:22:42 -0000	1.23.2.1
+++ iceboot/resources/startup.fs	21 Feb 2005 15:23:23 -0000
@@ -43,7 +43,7 @@
 
 \
 \ run domcal
-: echo-mode s" domcal" find if exec endif ;
+: dom-cal s" domcal" find if exec endif ;
 
 \
 \ comm stuff...
