whoops, inverted the new ADC scaler...

Index: hal/private/epxa10/hal/hal.c
===================================================================
RCS file: /home/icecube/cvsroot/hal/private/epxa10/hal/hal.c,v
retrieving revision 1.61.2.3
diff -u -r1.61.2.3 hal.c
--- hal/private/epxa10/hal/hal.c	2 Apr 2005 04:25:29 -0000	1.61.2.3
+++ hal/private/epxa10/hal/hal.c	5 Apr 2005 18:31:05 -0000
@@ -110,7 +110,7 @@
          sum += max1139Read(channel/12, channel % 12);
       }
    
-      return sum * 2048 / (3300 * n);
+      return sum * 3300 / (2048 * n);
    }
 #else
    return max1139Read(channel/12, channel % 12);
