Add "are you sure?" check to enableFBmin iceboot function

Index: sfi.c
===================================================================
RCS file: /home/icecube/cvsroot/iceboot/private/iceboot/sfi.c,v
retrieving revision 1.116.2.2
diff -u -r1.116.2.2 sfi.c
--- sfi.c	15 Mar 2005 22:56:41 -0000	1.116.2.2
+++ sfi.c	11 Apr 2005 15:52:28 -0000
@@ -1044,6 +1044,25 @@
 }
 
 static const char *enableFBmin(const char *p) {
+    
+    /* Warn users of risks... */
+    while (1) { 
+        char c;
+        int nr;
+        
+        printf("enableFBmin: LEDs may come on (steady-state) on older flasherboards.\r\n");
+        printf("Use enableFB unless CPLD state prevents it!  Continue [y/n]? ");
+        fflush(stdout);
+        
+        nr = read(0, &c, 1);
+        
+        if (nr==1) {
+            printf("%c\r\n", c); fflush(stdout);
+            if (toupper(c)=='Y') break;
+            else if (toupper(c)=='N') { return p; }
+        }
+    }
+    
     hal_FB_enable_min();
     return p;
 }
