fix ADC ch1 and 4 

Index: stf/private/stf-apps/ADC.c
===================================================================
RCS file: /home/icecube/cvsroot/stf/private/stf-apps/ADC.c,v
retrieving revision 1.10
retrieving revision 1.14
diff -u -r1.10 -r1.14
--- stf/private/stf-apps/ADC.c	20 Jul 2004 20:35:50 -0000	1.10
+++ stf/private/stf-apps/ADC.c	13 Apr 2005 22:27:58 -0000	1.14
@@ -1,5 +1,6 @@
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 
 #include <math.h>
 
@@ -13,7 +14,7 @@
 		 unsigned *pass_or_fail) {
 
    const int total_channel = 24;
-   unsigned i, j, adc_count, dac_count, sum;
+   unsigned i, j, dac_count;
    float dac_to_adc;
    memset(adc_value,0,total_channel*sizeof(unsigned));
    memset(pass_or_fail,0,total_channel*sizeof(unsigned));
@@ -37,7 +38,7 @@
        pass_or_fail[i]=(adc_count<=50) ? 1:0;            /* 65 counts = [-5V + (3.3V+5V)*162K/(100K+162K)]/0.002V */
      }
      else if (i == DOM_HAL_ADC_5V_POWER_SUPPLY ) {
-       pass_or_fail[i]=(adc_count<=1020 && adc_count>=980) ? 1:0;  /* 1000 counts = [0V + 5V*10K/(10K+15K)]/0.002V */
+       pass_or_fail[i]=(adc_count<=1030 && adc_count>=973) ? 1:0;  /* 1000 counts = [0V + 5V*10K/(10K+15K)]/0.002V */
      }
      else if (i == DOM_HAL_ADC_PRESSURE ) {
        pass_or_fail[i]=1; /* was: (dc_count<=870 && adc_count>=830) ? 1:0; default = 860, Azriel suggested to make this test pass all the time!*/        
@@ -46,7 +47,7 @@
        pass_or_fail[i]=(adc_count<=600 && adc_count>=400) ? 1:0;   /* default = 500 */                
      }
      else if (i == DOM_HAL_ADC_3_3V_CURRENT) {
-       pass_or_fail[i]=(adc_count<=115 && adc_count>=75) ? 1:0;    /* default = 96 */               
+       pass_or_fail[i]=(adc_count<=115 && adc_count>=72) ? 1:0;    /* default = 96 */               
      }
      else if (i == DOM_HAL_ADC_2_5V_CURRENT) {
        pass_or_fail[i]=(adc_count<=60 && adc_count>=0) ? 1:0;     /* default = 31 */              
