fix waveform position problem in atwd_pmt_spe

Index: stf/private/stf-apps/atwd_pmt_spe.c
===================================================================
RCS file: /home/icecube/cvsroot/stf/private/stf-apps/atwd_pmt_spe.c,v
retrieving revision 1.7.2.1
retrieving revision 1.10
diff -u -r1.7.2.1 -r1.10
--- stf/private/stf-apps/atwd_pmt_spe.c	22 Dec 2004 22:02:00 -0000	1.7.2.1
+++ stf/private/stf-apps/atwd_pmt_spe.c	6 Apr 2005 23:19:15 -0000	1.10
@@ -185,7 +185,7 @@
    free(buffer);
 
    return 
-      *atwd_waveform_position >= 9 &&
+      *atwd_waveform_position >= 6 &&
       *atwd_waveform_position <= 11 &&
       *atwd_waveform_amplitude > 15 &&
       *real_hv_output > 0.95*pmt_hv_high_volt &&
