Index: cpprod.sh
===================================================================
RCS file: /home/icecube/cvsroot/dom-ws/cpprod.sh,v
retrieving revision 1.6.2.1
retrieving revision 1.7
diff -u -r1.6.2.1 -r1.7
--- cpprod.sh	25 Jan 2005 10:31:07 -0000	1.6.2.1
+++ cpprod.sh	15 Apr 2005 18:11:55 -0000	1.7
@@ -13,19 +13,13 @@
 #
 # cp files and setup new link...
 #
-scp prod-${rev}-${num}.tar.gz getprod.sh glacier.lbl.gov:public_html/prod
 scp prod-${rev}-${num}.tar.gz \
   glacier.lbl.gov:/home/icecube/system/httpd/htdocs/releases/DOM-MB/stable_hex/
-ssh glacier.lbl.gov \
- "cd public_html/prod; rm -f prod-latest.tar.gz;\
- ln -s prod-${rev}-${num}.tar.gz prod-latest.tar.gz"
 
 ssh glacier.lbl.gov \
  "cd /home/icecube/system/httpd/htdocs/releases/DOM-MB/stable_hex; \
   rm -f prod-latest.tar.gz; \
   ln -s prod-${rev}-${num}.tar.gz prod-latest.tar.gz"
 
-scp ChangeLog glacier.lbl.gov:/home/icecube/system/httpd/htdocs/releases/DOM-MB/stable_hex
-
 exec ./tagprod.sh
 
