Index: private/dataAccess/dataAccessRoutines.c
===================================================================
RCS file: /home/icecube/cvsroot/testdomapp/private/dataAccess/dataAccessRoutines.c,v
retrieving revision 1.25.4.1.2.5
retrieving revision 1.54
diff -u -r1.25.4.1.2.5 -r1.54
--- private/dataAccess/dataAccessRoutines.c	15 Mar 2005 23:05:50 -0000	1.25.4.1.2.5
+++ private/dataAccess/dataAccessRoutines.c	10 May 2005 22:10:35 -0000	1.54
@@ -916,6 +916,7 @@
 #endif
       TEND(&breadout);
     } else {
+      MiscBits = 1;
       TBEG(breadout);
 #ifdef DOCH0
       hal_FPGA_TEST_readout(0, 0, 0, 0, Channel0Data, 0, 0, 0,
@@ -964,7 +965,7 @@
   }
   if(gottrig) { 
     /* Timestamp it */
-    if(FPGA_trigger_mode == TEST_DISC_TRIG_MODE) {
+    if(FPGA_trigger_mode == TEST_DISC_TRIG_MODE || FPGA_trigger_mode == FB_TRIG_MODE) {
       if(FPGA_ATWD_select == 0) 
 	time = hal_FPGA_TEST_get_atwd0_clock();
       else 
