Index: private/iceboot/sfi.c
===================================================================
RCS file: /home/icecube/cvsroot/iceboot/private/iceboot/sfi.c,v
retrieving revision 1.116.2.3
diff -u -r1.116.2.3 sfi.c
--- private/iceboot/sfi.c	14 Apr 2005 00:08:25 -0000	1.116.2.3
+++ private/iceboot/sfi.c	22 Apr 2005 18:23:00 -0000
@@ -1158,6 +1158,16 @@
     return p;
 }
 
+static const char *setFBrate(const char *p) {
+    if (hal_FB_isEnabled()) {
+        int value = pop();
+        hal_FPGA_TEST_FB_set_rate(value);
+    }
+    else
+        printf("Please power the flasherboard with enableFB first!\r\n");    
+    return p;
+}
+
 static int getFBdcdc(void) {
     int val = -1;
     if (hal_FB_isEnabled())
@@ -2647,6 +2657,7 @@
      { "startFBflashing", startFBflashing },
      { "stopFBflashing", stopFBflashing },
      { "setFBdcdc", setFBdcdc },
+     { "setFBrate", setFBrate },
 #if defined(PSKHACK)
      { "scan-angles", scanAngles },
      { "dump-cordic", dumpCordic },
Index: private/iceboot-docs/iceboot-ug.tex
===================================================================
RCS file: /home/icecube/cvsroot/iceboot/private/iceboot-docs/iceboot-ug.tex,v
retrieving revision 1.15.2.1
diff -u -r1.15.2.1 iceboot-ug.tex
--- private/iceboot-docs/iceboot-ug.tex	24 Nov 2004 19:48:19 -0000	1.15.2.1
+++ private/iceboot-docs/iceboot-ug.tex	22 Apr 2005 18:23:11 -0000
@@ -573,6 +573,8 @@
 \cmd{setFBmux}{channel}{}{ Select which LED current is muxed back to the
   mainboard.  An argument of 0 disables the mux, while 13 selects the TTL
   driver pulse. }
+\cmd{setFBrate}{rate}{}{ Set the rate (in Hz) for the flasherboard pulses.
+ Rates supported are 610, 305, 153, 76, 38, 19, 10, 5, 2, and 1 Hz. }
 \cmd{setFBwidth}{width}{}{ Set the flasherboard pulse width }
 \cmd{setFBdcdc}{}{enable}{ Enable/disable the DC/DC converter on the
   flasherboard which drives the LEDs }
