Index: private/dataAccess/compressEvent.c
===================================================================
RCS file: /home/icecube/cvsroot/testdomapp/private/dataAccess/compressEvent.c,v
retrieving revision 1.8.2.1
diff -u -r1.8.2.1 compressEvent.c
--- private/dataAccess/compressEvent.c	20 Nov 2004 20:55:16 -0000	1.8.2.1
+++ private/dataAccess/compressEvent.c	22 Apr 2005 18:24:29 -0000
@@ -328,7 +328,7 @@
 
 
     /* Compress all ATWD channels available */
-    int atwd = getEngATWDNumber(buf_in);
+    //int atwd = getEngATWDNumber(buf_in);
     for (ch = 0; ch < 4; ch++) {
         
         if ((data = getEngChannelData(buf_in, ch)) != NULL) {
