Index: private/stf-apps/ADC.c
===================================================================
RCS file: /home/icecube/cvsroot/stf/private/stf-apps/ADC.c,v
retrieving revision 1.10.2.1
retrieving revision 1.15
diff -u -r1.10.2.1 -r1.15
--- private/stf-apps/ADC.c	13 Apr 2005 23:33:07 -0000	1.10.2.1
+++ private/stf-apps/ADC.c	17 May 2005 17:36:30 -0000	1.15
@@ -47,7 +47,7 @@
        pass_or_fail[i]=(adc_count<=600 && adc_count>=400) ? 1:0;   /* default = 500 */                
      }
      else if (i == DOM_HAL_ADC_3_3V_CURRENT) {
-       pass_or_fail[i]=(adc_count<=115 && adc_count>=72) ? 1:0;    /* default = 96 */               
+       pass_or_fail[i]=(adc_count<=115 && adc_count>=63) ? 1:0;    /* default = 96 */               
      }
      else if (i == DOM_HAL_ADC_2_5V_CURRENT) {
        pass_or_fail[i]=(adc_count<=60 && adc_count>=0) ? 1:0;     /* default = 31 */              
