Index: cpprod.sh
===================================================================
RCS file: /home/icecube/cvsroot/dom-ws/cpprod.sh,v
retrieving revision 1.6.2.2
retrieving revision 1.8
diff -u -r1.6.2.2 -r1.8
--- cpprod.sh	16 May 2005 16:55:05 -0000	1.6.2.2
+++ cpprod.sh	17 May 2005 17:41:48 -0000	1.8
@@ -21,5 +21,7 @@
   rm -f prod-latest.tar.gz; \
   ln -s prod-${rev}-${num}.tar.gz prod-latest.tar.gz"
 
+scp ChangeLog glacier.lbl.gov:/home/icecube/system/httpd/htdocs/releases/DOM-MB/stable_hex
+
 exec ./tagprod.sh
 
