Index: cpprod.sh
===================================================================
RCS file: /home/icecube/cvsroot/dom-ws/cpprod.sh,v
retrieving revision 1.6.2.3
diff -u -r1.6.2.3 cpprod.sh
--- cpprod.sh	27 May 2005 00:44:03 -0000	1.6.2.3
+++ cpprod.sh	8 Jun 2005 07:50:17 -0000
@@ -13,15 +13,12 @@
 #
 # cp files and setup new link...
 #
-scp prod-${rev}-${num}.tar.gz \
-  glacier.lbl.gov:/home/icecube/system/httpd/htdocs/releases/DOM-MB/stable_hex/
+scp ChangeLog prod-${rev}-${num}.tar.gz \
+  glacier.lbl.gov:/var/www/html/releases/DOM-MB/stable_hex
 
 ssh glacier.lbl.gov \
- "cd /home/icecube/system/httpd/htdocs/releases/DOM-MB/stable_hex; \
+ "cd /var/html/releases/DOM-MB/stable_hex; \
   rm -f prod-latest.tar.gz; \
   ln -s prod-${rev}-${num}.tar.gz prod-latest.tar.gz"
 
-scp ChangeLog glacier.lbl.gov:/home/icecube/system/httpd/htdocs/releases/DOM-MB/stable_hex
-
 exec ./tagprod.sh
-
