Index: iceboot/private/iceboot/sfi.c
===================================================================
RCS file: /home/icecube/cvsroot/iceboot/private/iceboot/sfi.c,v
retrieving revision 1.116.2.4
diff -u -r1.116.2.4 sfi.c
--- iceboot/private/iceboot/sfi.c	16 May 2005 21:43:49 -0000	1.116.2.4
+++ iceboot/private/iceboot/sfi.c	1 Jul 2005 14:16:53 -0000
@@ -1630,7 +1630,7 @@
 
 static int readPressure(void) {  
 
-    unsigned int loop_cnt = 1000;
+    unsigned int loop_cnt = 50;
     unsigned long pressure_sum = 0;
     unsigned long voltage_sum = 0;
     float mean_pressure, mean_voltage;
