Index: src/icecube/daq/stf/autogen/STFParameterLookup.java
===================================================================
RCS file: /home/icecube/cvsroot/stf/src/icecube/daq/stf/autogen/STFParameterLookup.java,v
retrieving revision 1.7.4.6
diff -u -r1.7.4.6 STFParameterLookup.java
--- src/icecube/daq/stf/autogen/STFParameterLookup.java	8 Jun 2005 07:47:49 -0000	1.7.4.6
+++ src/icecube/daq/stf/autogen/STFParameterLookup.java	6 Oct 2005 20:32:58 -0000
@@ -271,10 +271,13 @@
     hm.put("hv_id", p);
     p = new STFParameter( false );
     p.setType(STFParameterType.getType("unsignedInt"));
+    hm.put("hv_max_diff_mv", p);
+    p = new STFParameter( false );
+    p.setType(STFParameterType.getType("unsignedInt"));
     hm.put("hv_worst_set_mvolt", p);
     p = new STFParameter( false );
     p.setType(STFParameterType.getType("unsignedInt"));
-    hm.put("hv_worst_read_mvolt", p);
+    hm.put("hv_avg_diff_mv", p);
 
     ret.put("pmt_hv_stability", hm=new HashMap());
     p = new STFParameter( true );
@@ -768,6 +771,9 @@
     p.setType(STFParameterType.getType("unsignedInt"));
     hm.put("min_slope_led", p);
     p = new STFParameter( false );
+    p.setType(STFParameterType.getType("unsignedInt"));
+    hm.put("failing_led_cnt", p);
+    p = new STFParameter( false );
     p.setType(STFParameterType.getType("unsignedIntArray"));
     hm.put("led_avg_current", p);
 
@@ -850,6 +856,9 @@
     p = new STFParameter( false );
     p.setType(STFParameterType.getType("unsignedInt"));
     hm.put("failing_led", p);
+    p = new STFParameter( false );
+    p.setType(STFParameterType.getType("unsignedInt"));
+    hm.put("failing_led_cnt", p);
     p = new STFParameter( false );
     p.setType(STFParameterType.getType("unsignedIntArray"));
     hm.put("led_avg_current", p);
