Index: private/slowControl/domSControl.c
===================================================================
RCS file: /home/icecube/cvsroot/testdomapp/private/slowControl/domSControl.c,v
retrieving revision 1.16.6.3
retrieving revision 1.24
diff -u -r1.16.6.3 -r1.24
--- private/slowControl/domSControl.c	15 Mar 2005 23:05:50 -0000	1.16.6.3
+++ private/slowControl/domSControl.c	19 Oct 2005 20:32:21 -0000	1.24
@@ -474,20 +474,26 @@
 		   Message_setStatus(M,SUCCESS);
 		 } else if(data[0] == 1) {
 		   LCmode = data[0];
-		   hal_FPGA_TEST_enable_spe_lc(1,1);
+		   hal_FPGA_TEST_enable_spe_lc(1, 1, DOM_HAL_LC_LOGIC_OR);
                    moniInsertLCModeChangeMessage(hal_FPGA_TEST_get_local_clock(),
                                                  LCmode);
 		   Message_setStatus(M,SUCCESS);
                  } else if(data[0] == 2) { /* Upper ONLY */
                    LCmode = data[0];
-                   hal_FPGA_TEST_enable_spe_lc(0, 1);
+                   hal_FPGA_TEST_enable_spe_lc(0, 1, DOM_HAL_LC_LOGIC_OR);
                    moniInsertLCModeChangeMessage(hal_FPGA_TEST_get_local_clock(),
                                                  LCmode);
                    Message_setStatus(M,SUCCESS);
                  } else if(data[0] == 3) { /* Lower ONLY */
                    LCmode = data[0];
-                   hal_FPGA_TEST_enable_spe_lc(1, 0);
+                   hal_FPGA_TEST_enable_spe_lc(1, 0, DOM_HAL_LC_LOGIC_OR);
                    moniInsertLCModeChangeMessage(hal_FPGA_TEST_get_local_clock(),
+                                                 LCmode);
+                   Message_setStatus(M,SUCCESS);
+		 } else if(data[0] == 4) { /* up AND down */
+		   LCmode = data[0];
+                   hal_FPGA_TEST_enable_spe_lc(1, 1, DOM_HAL_LC_LOGIC_AND);
+		   moniInsertLCModeChangeMessage(hal_FPGA_TEST_get_local_clock(),
                                                  LCmode);
                    Message_setStatus(M,SUCCESS);
 		 } else {
